### PR TITLE
BottleLoader: Fix installing a bottle from an URL

### PR DESCRIPTION
**Pull request by sjackman**
The name of the formula is not extracted correctly when the URL includes a hyphen.

-----

- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [x] Have you successfully run `brew tests` with your changes locally?

### CODE REVIEW

File: Library/Homebrew/formulary.rb [hunk 1]
```diff
@@ -108,7 +108,7 @@ def initialize(bottle_name)
       case bottle_name
       when %r{(https?|ftp|file)://}
         # The name of the formula is found between the last slash and the last hyphen.
-        resource = Resource.new bottle_name[%r{([^/]+)-}, 1] { url bottle_name }
+        resource = Resource.new File.basename(bottle_name)[/(.+)-/, 1] { url bottle_name }
```
Comment (MikeMcQuaid) [inline Library/Homebrew/formulary.rb]: Can we split this onto 2-3 lines, it's pretty hard to follow all the parens as-is. Thanks!

Comment (sjackman) [inline Library/Homebrew/formulary.rb]: ✅ Done!

**Comment by MikeMcQuaid** [comment]
Thanks @sjackman 1

**Comment by sjackman** [comment]
Thanks for merging, Mike!
